### PR TITLE
*: Start switching to the zap logger

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
In order to smoothly switch to the new logger

### What is changed and how it works?
- Replace logger with `pingcap/log` global logger
- Use customer zap logger in `etcd `

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: cmd/pd-server/main.go [hunk 1]
```diff
@@ -51,18 +51,29 @@ func main() {
...
+	// New zap logger
+	err = cfg.SetupLogger()
+	if err != nil {
+		log.L().Fatal("initialize logger error", zap.Error(err))
+	}
+	// Flushing any buffered log entries
+	defer func() {
+		log.S().Sync()
+		log.L().Sync()
+	}()
```
Comment (disksing) [inline cmd/pd-server/main.go]: I don't think `defer func` will be called after `os.Exit`

File: server/config.go [hunk 2]
```diff
@@ -62,7 +64,7 @@ type Config struct {
 	LeaderLease int64 `toml:"lease" json:"lease"`
 
 	// Log related config.
-	Log logutil.LogConfig `toml:"log" json:"log"`
+	Log log.Config `toml:"log" json:"log"`
```
Comment (disksing) [inline server/config.go]: Is it guaranteed to be compatible with the original configuration file?

File: server/config.go [hunk 3]
```diff
@@ -772,6 +777,18 @@ func ParseUrls(s string) ([]url.URL, error) {
...
 
+// SetupLogger setup the logger.
+func (c *Config) SetupLogger() error {
+	lg, syncer, err := log.InitLogger(&c.Log)
+	if err != nil {
+		return err
+	}
+	c.logger = lg
+	c.logSyncer = syncer
+	log.ReplaceGlobals(lg)
```
Comment (disksing) [inline server/config.go]: It looks a bit strange here. I think `InitLogger` and `ReplaceGlobal` are more suitable in the main function.

File: server/config.go (hunk 2 above)
Comment (nolouch) [reply server/config.go]: Yes.

File: cmd/pd-server/main.go [hunk 4]
```diff
@@ -16,18 +16,18 @@ package main
...
-	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	log "github.com/pingcap/log"
 	"github.com/pingcap/pd/pkg/logutil"
 	"github.com/pingcap/pd/pkg/metricutil"
 	"github.com/pingcap/pd/server"
 	"github.com/pingcap/pd/server/api"
 	"github.com/pkg/errors"
-	log "github.com/sirupsen/logrus"
+	"go.uber.org/zap"
```
Comment (siddontang) [inline cmd/pd-server/main.go]: can we hide all zap in `pingcap/log`?

File: server/testutil.go [hunk 5]
```diff
@@ -56,6 +58,8 @@ func NewTestServer(c *check.C) (*Config, *Server, CleanupFunc, error) {
 	return cfg, s, cleanup, nil
 }
 
+var zapLogOnce = new(sync.Once)
```
Comment (rleungx) [inline server/testutil.go]: Why do we need to use `new` here?

File: server/util.go [hunk 6]
```diff
@@ -76,7 +77,10 @@ func CheckPDVersion(opt *scheduleOption) {
 	}
 	clusterVersion := opt.loadClusterVersion()
 	if pdVersion.LessThan(clusterVersion) {
-		log.Warnf("PD version %s less than cluster version: %s, please upgrade PD", pdVersion, clusterVersion)
+		log.Warn(
+			"PD version less than cluster version, please upgrade PD",
+			zap.String("PD Version", pdVersion.String()),
```
Comment (rleungx) [inline server/util.go]: How about using a hyphen here?

**Pull request merged**